packages/framework-core/src/services/read-model-store.ts

```diff
@@ -182,12 +364,102 @@ export class ReadModelStore {
     return undefined
   }
 
+  public async callFunction<TReadModel extends ReadModelInterface>(
+    projectionMetadata: ProjectionMetadata<EntityInterface, ReadModelInterface>,
+    entity: EntityInterface,
+    migratedReadModel: ReadModelInterface | undefined,
+    readModelID: UUID | undefined
+  ): Promise<ProjectionResult<TReadModel> | undefined> {
+    try {
+      const projectionMetadataJoinKey = projectionMetadata.joinKey
+      const projectionFunction = this.getProjectionFunction(projectionMetadata)
+      if (this.isJoinKeyByEntity(projectionMetadataJoinKey)) {
+        return Array.isArray(entity[projectionMetadataJoinKey])
+          ? projectionFunction(entity, readModelID, migratedReadModel || null)
+          : projectionFunction(entity, migratedReadModel || null)
+      }
+      return projectionFunction(entity, readModelID, migratedReadModel || null)
+    } catch (e) {
+      const globalErrorDispatcher = new BoosterGlobalErrorDispatcher(this.config)
+      const error = await globalErrorDispatcher.dispatch(new ProjectionGlobalError(entity, migratedReadModel, e))
+      if (error) throw error
+    }
+    return undefined
+  }
+
   // eslint-disable-next-line @typescript-eslint/ban-types
-  public projectionFunction(projectionMetadata: ProjectionMetadata<EntityInterface>): Function {
+  getProjectionFunction(projectionMetadata: ProjectionMetadata<EntityInterface, ReadModelInterface>): Function {
```

Comment:
I guess that this new name it is better, since it returns a Function. I guess that if this is not called elsewhere, it is good to have it non-public

Reply:
This one can't be made non-public since it's being called in some unit tests.